include/openenclave/bits/sgx/seal.h

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#ifndef _OE_BITS_SGX_SEAL_H
+#define _OE_BITS_SGX_SEAL_H
+
+#include <openenclave/bits/sgx/sgxtypes.h>
+
+OE_EXTERNC_BEGIN
+
+/**
+ * SGX specific seal settings
+ */
+typedef enum _oe_seal_setting_name_sgx
+{
+    OE_SEAL_SETTING_SGX_KEYNAME = 0x100,
+    OE_SEAL_SETTING_SGX_ISVSVN,
+    OE_SEAL_SETTING_SGX_CET_ATTRIBUTES_MASK,
+    OE_SEAL_SETTING_SGX_CPUSVN,
+    OE_SEAL_SETTING_SGX_ATTREMASK,
+    OE_SEAL_SETTING_SGX_XFRMMASK,
+    OE_SEAL_SETTING_SGX_MISCMASK,
+    OE_SEAL_SETTING_SGX_CONFIGSVN,
+    OE_SEAL_SETTING_SGX_MAX
+} oe_seal_setting_name_sgx_t;
+
+#define _DEFINE_SEAL_SETTING_SGX(name, type)   \
+    typedef struct _oe_seal_setting_sgx_##name \
+    {                                          \
+        oe_seal_setting_header_t header;       \
+        type name;                             \
+    } oe_seal_setting_sgx_##name##_t
+
+_DEFINE_SEAL_SETTING_SGX(keyname, uint16_t);
+_DEFINE_SEAL_SETTING_SGX(isvsvn, uint16_t);
+_DEFINE_SEAL_SETTING_SGX(cet_attributes_mask, uint8_t);
+_DEFINE_SEAL_SETTING_SGX(attrmask, uint64_t);
+_DEFINE_SEAL_SETTING_SGX(xfrmmask, uint64_t);
+_DEFINE_SEAL_SETTING_SGX(miscmask, uint32_t);
+_DEFINE_SEAL_SETTING_SGX(configsvn, uint16_t);
+
+typedef struct _oe_seal_setting_sgx_cpusvn
+{
+    oe_seal_setting_header_t header;
+    uint8_t cpusvn[SGX_CPUSVN_SIZE];
+} oe_seal_setting_sgx_cpusvn_t;
+
+#undef _DEFINE_SEAL_SETTING_SGX
+
+typedef struct _oe_seal_plugin_definition_sgx
+{
+    const oe_uuid_t id;
+
+    oe_result_t (*seal_callback)(
+        sgx_key_request_t* keyrequest,
+        const oe_seal_setting_header_t* settings[],
+        size_t settings_count,
+        const uint8_t* plaintext,
+        size_t plaintext_size,
+        uint8_t** blob,
+        size_t* blob_size);
+
+    oe_result_t (*unseal_callback)(
+        const uint8_t* blob,
+        size_t blob_size,
+        uint8_t** plaintext,
+        size_t* plaintext_size);
+} oe_seal_plugin_definition_sgx_t;
+
+oe_result_t oe_register_seal_plugin(
+    const oe_seal_plugin_definition_sgx_t* plugin,
```

Comment:
As discussed in the SIG-Arch meeting, use void* and move to non-sgx file

Reply:
As I responded in SIG-Arch meeting today, the intention here is for strong type checking, hence a plug-in developed for one TEE cannot (i.e., would result in a compiler error) be registered under a different TEE. Making it typeless (`void*`) is simple but will shift the error checking to runtime, which IMHO is a big disadvantage. 

But I understand your point that it's more important to have a TEE-agnostic API. I'll make the changes.